system/lib/wasmfs/syscalls.cpp

```diff
@@ -465,6 +465,12 @@ static __wasi_fd_t doOpen(path::ParsedParent parsed,
     return -EEXIST;
   }
 
+  // If O_TRUNC, truncate the file if possible.
+  if (flags & O_TRUNC && child->is<DataFile>() &&
+      fileMode & WASMFS_PERM_WRITE) {
```

Comment:
Do we not need `()` around the `&` operation and its arguments? At least in JS I believe we do. For clarity I'd add it here.

Also, if we lack write permissions should we not error?

Reply:
I can add parens, although my compiler didn’t mind me leaving them out. O_TRUNC is ignored if the file is not writable.